### PR TITLE
Feature/backbone specific addresses

### PR DESCRIPTION
**Pull request by jkoenig134**

### CODE REVIEW

**Comment by jkoenig134** [comment]
created a draft to hold this branch up2date

cc @tnotheis

**Comment by jkoenig134** [comment]
The current occuring error ist caused by the backbone having some identity-problems, thinking it is prod.enmeshed.eu :D @tnotheis ??? :D

File: .vscode/settings.json [hunk 1]
```diff
@@ -49,10 +49,7 @@
     "xmlTools.splitAttributesOnFormat": true,
     "xmlTools.enforcePrettySelfClosingTagOnFormat": true,
     "editor.codeActionsOnSave": {
-        "source.organizeImports": "explicit"
+        "source.organizeImports": "always"
```
Comment (jkoenig134) [inline .vscode/settings.json:52]: @sebbi08 this fixes our imports problems :(

File: packages/runtime/src/useCases/common/validation/ValidatableStrings.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /**
- * @pattern id1[A-Za-z0-9]{32,33}
+ * @pattern did:e:[a-zA-Z\.]*:dids:[0-9a-f]{22}
```
Comment (slavistan) [inline packages/runtime/src/useCases/common/validation/ValidatableStrings.ts:2]: `did:e:[A-Za-z0-9.-]+:dids:[0-9a-f]{22}`

The domain part should allow for dashes and digits (e.g. _did:e:preprod-0.enmeshed.eu_), and shouldn't be optional (right?). Also, inside a character set `[...]` the period character is interpreted literally, so you don't need to escape it.

Comment (jkoenig134) [reply packages/runtime/src/useCases/common/validation/ValidatableStrings.ts]: Good catch!

**Review by tnotheis — APPROVED** [review]

File: packages/transport/src/modules/accounts/IdentityUtil.ts [hunk 3]
```diff
@@ -1,56 +1,53 @@
-import { CoreBuffer, CryptoHash, CryptoHashAlgorithm, ICryptoSignaturePublicKey } from "@nmshd/crypto";
-import { CoreAddress, CoreErrors } from "../../core";
+import { CoreBuffer, CryptoHash, CryptoHashAlgorithm, Encoding, ICryptoSignaturePublicKey } from "@nmshd/crypto";
+import { CoreAddress } from "../../core";
 
-export class IdentityUtil {
-    public static async createAddress(publicKey: ICryptoSignaturePublicKey, realm: string): Promise<CoreAddress> {
-        if (realm.length !== 3) throw CoreErrors.general.realmLength();
+const enmeshedPrefix = "did:e:";
```
Comment (slavistan) [inline packages/transport/src/modules/accounts/IdentityUtil.ts:4]: Vague name. How about `enmeshedAddressPrefix`?

Comment (jkoenig134) [reply packages/transport/src/modules/accounts/IdentityUtil.ts:4]: yeah, i also thought about inlining it completely. As it's only used in this file I don't care about the name that much. will change.

**Review by tnotheis — APPROVED** [review]

File: packages/transport/src/modules/accounts/IdentityUtil.ts (hunk 3 above)
Comment (jkoenig134) [reply packages/transport/src/modules/accounts/IdentityUtil.ts]: I already changed the variable name :D

**Pull request merged**